### PR TITLE
WIP JAMES-2586 PostgresMailboxAnnotationDAO and Mapper

### PR DESCRIPTION
**Pull request by Arsnael**
Just started last week on it... Not much done yet

### CODE REVIEW

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/dao/PostgresMailboxAnnotationDAO.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            .collect(Collectors.toList());
+    }
+
+    public Mono<Void> insertAnnotation(PostgresMailboxId mailboxId, MailboxAnnotation mailboxAnnotation) {
+        Preconditions.checkArgument(!mailboxAnnotation.isNil());
+
+        // TODO: likely wrong... might need to check if row exists with mailboxId and then insert if not, update if is
+        return postgresExecutor.executeVoid(dslContext ->
+            Mono.from(dslContext.insertInto(TABLE_NAME, MAILBOX_ID, ANNOTATIONS)
+                .values(mailboxId.asUuid(), annotationAsHstore(mailboxAnnotation))));
```
Comment (chibenwa) [inline mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/dao/PostgresMailboxAnnotationDAO.java:71]: Here we use HStore on a mutable field. 
Thus we likely need in place updates.

```
UPDATE votre_table
SET votre_colonne_hstore = votre_colonne_hstore || 'nouvelle_cle' => 'nouvelle_valeur'
WHERE votre_condition;

UPDATE votre_table
SET votre_colonne_hstore = votre_colonne_hstore - 'cle_a_retirer'
WHERE votre_condition;
```

```
dsl.update(table("votre_table"))
   .set(field("votre_colonne_hstore"), field("votre_colonne_hstore").concat(field("'nouvelle_cle => nouvelle_valeur'")))
   .where(votre_condition)
   .execute();
   
dsl.update(table("votre_table"))
   .set(field("votre_colonne_hstore"), field("votre_colonne_hstore").minus(field("'cle_a_retirer'")))
   .where(votre_condition)
   .execute();
```

Comment (Arsnael) [reply mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/dao/PostgresMailboxAnnotationDAO.java:71]: Not only cause we have the case when the line with mailboxId does not exist yet, so we need to do an insert. Hung and Quan did something like that with quota DAOs, and I intend to reuse something similar

**Comment by hungphan227** [comment]
new PR https://github.com/apache/james-project/pull/1822

**Comment by Arsnael** [comment]
Let me close this then.

**Pull request closed without merging**